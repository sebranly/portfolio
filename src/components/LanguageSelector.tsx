import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { languages } from '../languages';
import { Language } from '../types';

export interface LanguageSelectorProps {}

const LanguageSelector: React.FC<LanguageSelectorProps> = (props) => {
  const { i18n } = useTranslation();

  return (
    <div className="mb-4">
      {languages.map((language: Language) => {
        const { shortName, name } = language;
        const isSelected = i18n.resolvedLanguage === shortName;

        return (
          <button
            className="rounded mr-2"
            key={shortName}
            style={{
              fontWeight: isSelected ? 'bold' : 'normal',
              textDecorationLine: isSelected ? 'underline' : 'none'
            }}
            type="submit"
            onClick={() => i18n.changeLanguage(shortName)}
          >
            {name}
          </button>
        );
      })}
    </div>
  );
};

export { LanguageSelector };
