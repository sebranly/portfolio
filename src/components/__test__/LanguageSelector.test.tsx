import * as React from 'react';
import { render } from '@testing-library/react';

import { LanguageSelector } from '../LanguageSelector';

test('LanguageSelector', () => {
  const createComponent = () => <LanguageSelector />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
