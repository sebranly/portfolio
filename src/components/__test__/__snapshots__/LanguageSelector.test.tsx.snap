// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector 1`] = `
NodeList [
  <div
    class="mb-4"
  >
    <button
      class="rounded mr-2"
      style="font-weight: normal; text-decoration-line: none;"
      type="submit"
    >
      English
    </button>
    <button
      class="rounded mr-2"
      style="font-weight: normal; text-decoration-line: none;"
      type="submit"
    >
      Fran√ßais
    </button>
  </div>,
]
`;
